src/libraries/System.Text.Json/tests/Serialization/EnumConverterTests.cs

```diff
@@ -294,5 +295,230 @@ public enum MyEnum
+                for (int i = 1; i < 60; i++)
+                {
+                    dictionary = BuildDictionary(i);
+                    JsonSerializer.Serialize(dictionary, options);
+                }
+
+                dictionary = BuildDictionary(0);
+                string json = JsonSerializer.Serialize(dictionary, options);
+                Assert.Equal($"{{\"0\":0}}", json);
+            }
+        }
+
+        public class Int32EnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumInt32>
+        {
+            public override Dictionary<SampleEnumInt32, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumInt32, int> { { (SampleEnumInt32)i, i } };
+        }
+
+        public class UInt32EnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumUInt32>
+        {
+            public override Dictionary<SampleEnumUInt32, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumUInt32, int> { { (SampleEnumUInt32)i, i } };
+        }
+
+        public class UInt64EnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumUInt64>
+        {
+            public override Dictionary<SampleEnumUInt64, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumUInt64, int> { { (SampleEnumUInt64)i, i } };
+        }
+
+        public class Int64EnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumInt64>
+        {
+            public override Dictionary<SampleEnumInt64, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumInt64, int> { { (SampleEnumInt64)i, i } };
+        }
+
+        public class Int16EnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumInt16>
+        {
+            public override Dictionary<SampleEnumInt16, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumInt16, int> { { (SampleEnumInt16)i, i } };
+        }
+
+        public class UInt16EnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumUInt16>
+        {
+            public override Dictionary<SampleEnumUInt16, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumUInt16, int> { { (SampleEnumUInt16)i, i } };
+        }
+
+        public class ByteEnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumByte>
+        {
+            public override Dictionary<SampleEnumByte, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumByte, int> { { (SampleEnumByte)i, i } };
+        }
+
+        public class SByteEnumDictionary : NumericEnumKeyDictionaryBase<SampleEnumSByte>
+        {
+            public override Dictionary<SampleEnumSByte, int> BuildDictionary(int i) =>
+                new Dictionary<SampleEnumSByte, int> { { (SampleEnumSByte)i, i } };
+        }
+
+
+        [Flags]
+        public enum SampleEnumInt32
+        {
+            A = 1 << 0,
+            B = 1 << 1,
+            C = 1 << 2,
+            D = 1 << 3,
+            E = 1 << 4,
+            F = 1 << 5,
+            G = 1 << 6,
+            H = 1 << 7,
```

Comment:
Seems like `H` is never used by `SerilizeDictionaryWhenCacheIsFull`, you should remove it here and in the other enums.